pgmpy/models/BayesianModel.py

```diff
@@ -1030,3 +1030,16 @@ def get_markov_blanket(self, node):
         blanket_nodes = set(blanket_nodes)
         blanket_nodes.remove(node)
         return list(blanket_nodes)
+    
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            network_struct1 = (self.nodes(), self.edges())
+            network_struct2 = (other.nodes(), other.edges())
```

Comment:
@Vaibhavdixit02 I don't think this would work. Since networkx implements the `node` and `edge` properties as dict, you can't be sure of the order in which it's keys (which is what `nodes()` and `edges()` method returns) are returned. Better to compare the properties directly. Also, there's no need to create new tuples here. You can just compare the things in the `if` statement below. Something like:
```python
if self.node == other.node and self.edge == other.edge and other_conditions
```